### ISSUE TITLE
[HOLD - #11768] [$500] Unable to open chat through search when the browser/app size is decreased - reported by @adeel0202

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#9734]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/docs/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Decrease the width of your browser window to the point in which the chat view is not visible, only the LHN 
2. Click on the magnifying glass search icon in the LHN header
3. Click on any chat from the search results and observe that you're navigated back to a view of the LHN and not the view of the chat you just clicked on

## Expected Result:
When the screen width on Web/desktop has been decreased to the size of mobile/tablet, it should work in exactly the same way as those devices.

1. Click on the search icon to navigate to the search page
2. Click on any chat from the search results
3. Navigate to the chat view
4. Click the back caret `<` in the top left of the chat to return to the LHN  

## Actual Result:
The LHN is in view and not the chat

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- Desktop App

**Version Number:** 1.1.79-11
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 

https://user-images.githubusercontent.com/43995119/177471367-9f56c588-9160-4c85-8726-09282387d344.mov

**Upwork URL:** https://www.upwork.com/jobs/~010e7681ab13edf3b1
**Issue reported by:** @adeel0202
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1655398548680369

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by strepanier03** [comment]
Triaging now.

**Comment by strepanier03** [comment]
Hmm, I'll send it on but I can't repro myself in Chrome or Firefox.

**Comment by Gonals** [comment]
Good candidate for external!

**Comment by trjExpensify** [comment]
Ah yeah, I can repro this. I misread it initially, but yes.. accessing a chat from the `Search` page doesn't work. Job on upwork here: https://www.upwork.com/jobs/~010e7681ab13edf3b1

**Comment by Luke9389** [comment]
I think this issue needs a better description of what we want to happen. How would this look? How would you get back to the LHN view from the chat? What new components would we need?

All of that should be answered before we start taking proposals.

**Comment by trjExpensify** [comment]
Happy to answer point one and two, Luke. I've updated the steps and expected results to the below for additional clarity. For the third, perhaps you can pitch in on that as the assigned CME if you think something new is warranted here.

**Action Performed:**
1. Decrease the width of your browser window to the point in which the chat view is not visible, only the LHN 
2. Click on the magnifying glass search icon in the LHN header
3. Click on any chat from the search results and observe that you're navigated back to a view of the LHN and not the view of the chat you just clicked on

**Expected results:**

When the screen width on Web/desktop has been decreased to the size of mobile/tablet, it should work in exactly the same way as those devices.

1. Click on the search icon to navigate to the search page
2. Click on any chat from the search results
3. Navigate to the chat view
4. Click the back caret `<` in the top left of the chat to return to the LHN

**Comment by Luke9389** [comment]
Ok great! Thanks for those clarifiers @trjExpensify. 🙇 
Now that I see we're shooting for behavior that looks the same as the mobile layout, I don't think we'll need new components.

**Comment by trjExpensify** [comment]
Great, so we're now waiting on proposals here! 🎉

**Comment by trjExpensify** [comment]
Price doubled on Upwork here: https://www.upwork.com/jobs/~010e7681ab13edf3b1

**Comment by mollfpr** [comment]
### Proposal

This happens because when we open on the desktop layout the drawer state is set to `closed` and when we navigate from `stack` screen we set the drawer history to have the status `open`, while if we want to show the chat screen we should set the drawer status it to `closed`
https://github.com/Expensify/App/blob/97eb8bb0019068b484035a17a64e4ecdeb147092/src/libs/Navigation/DeprecatedCustomActions.js#L108-L115

We can solve this by setting the correct drawer status on`DeprecatedCustomActions` but I think @marcaaron will have this update from the internal side.

The other way is calling `dismissModal` whenever we navigate from `stack` to `drawer`, so the drawer history will not be set manually from custom actions.
https://github.com/Expensify/App/blob/97eb8bb0019068b484035a17a64e4ecdeb147092/src/libs/Navigation/Navigation.js#L137-L140
```diff
    if (isDrawerRoute(route)) {
+        // Check if the source is from other than drawer route
+        // eslint-disable-next-line no-use-before-define
+        if (!isDrawerRoute(getActiveRoute())) {
+            // eslint-disable-next-line no-use-before-define
+            dismissModal();
+        }

        navigationRef.current.dispatch(DeprecatedCustomActions.pushDrawerRoute(route));
        return;
    }
```

### Result

https://user-images.githubusercontent.com/25520267/183277710-8325936a-6550-4395-a962-9db453c91b40.mp4

**Comment by Santhosh-Sellavel** [comment]
The proposal is from `DeprecatedCustomActions.js` which is the unaccepted solution for now.

@trjExpensify Can you make it to weekly/monthly thanks!

**Comment by trjExpensify** [comment]
I caught up with Marc earlier and I'm going to pop the issue on `[Hold]` & `Planning` for the time being. We have plans to revisit `react-navigation` more holistically, so we should account for this issue then to avoid adding any additional workarounds or hacks atop of what is already complex code.

**Comment by trjExpensify** [comment]
This is held, Melv.

**Comment by Santhosh-Sellavel** [comment]
on hold

**Comment by trjExpensify** [comment]
^^ 👋 @Luke9389, has this come off hold or something?

**Comment by Luke9389** [comment]
OH! huh. Weeeird. I coulda sworn someone commented that this was off hold. 

My bad!

**Comment by trjExpensify** [comment]
Ha, all good!

**Comment by JmillsExpensify** [comment]
snagging this as I'm on the tracking issue.

**Comment by JmillsExpensify** [comment]
See the tracking issue for the latest.

**Comment by JmillsExpensify** [comment]
Still holding.

**Comment by JmillsExpensify** [comment]
Still held as this initiative goes through the design process.

**Comment by JmillsExpensify** [comment]
Still holding on the larger react navigation reboot.

**Comment by tgolen** [comment]
Is this really a case that we should be supporting? I think we only care about "normally" sized browser windows and mobile web clients. My guess is that at these widths, there is quite a bit more broken than just this navigation issue, and so supporting this browser platform and window size should be regarded as a new feature and not a bug.

**Comment by JmillsExpensify** [comment]
That's a really great point. I think you're right – to the extent that this also fails the "supported platform" stipulation for bugs as well. I'm actually going to close as a result but anyone from Expensify on this issue can re-open if they feel strongly.
